helpers/spark-shell.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import os
+import pwd
+import logging
+import argparse
+import utils
+
+USER_HOME_DIR_ENT_IDX = 5
+
+if __name__ == "__main__":
+    logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.INFO)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--master", default=None, type=str, help='Kubernetes control plane uri.')
+    parser.add_argument("--deploy-mode", default="client", type=str, help='Deployment mode for spark shell. Default is \'client\'.')
```

Comment:
Uhm, not sure we should allow the user to change this. Beside whatever he inputs, it would not be taken from the application. So, I'd just skip this argument

Reply:
Actually, if we do not explicitly parse this out, it would go to un-parsed extra_args where it would be more difficult to change it from cluster to client for example. hence, I thought of parsing it away and then overriding it explicitly to client every time.

If you notice below, I never actually consult or use args.deploy_mode at all in the final spark-shell command.